thrift_src/CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+# Find Thrift
+find_package(Thrift REQUIRED)
+
+# Include directories
+include_directories(
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${CMAKE_CURRENT_BINARY_DIR}
+  ${THRIFT_INCLUDE_DIRS}
+)
+
+# Get Thrift files
+file(GLOB THRIFT_FILES "${CMAKE_CURRENT_SOURCE_DIR}/*.thrift")
+
+# Generate Thrift sources
+set(THRIFT_GEN_CPP_DIR ${CMAKE_CURRENT_BINARY_DIR}/gen-cpp)
+set(THRIFT_GEN_PY_DIR ${CMAKE_CURRENT_BINARY_DIR}/gen-py)
+
+# Create directories
+file(MAKE_DIRECTORY ${THRIFT_GEN_CPP_DIR})
+file(MAKE_DIRECTORY ${THRIFT_GEN_PY_DIR})
+
+# Generate Thrift sources
+foreach(THRIFT_FILE ${THRIFT_FILES})
+  get_filename_component(THRIFT_NAME ${THRIFT_FILE} NAME_WE)
+  
+  set(THRIFT_CPP_FILES
+    ${THRIFT_GEN_CPP_DIR}/${THRIFT_NAME}_constants.cpp
+    ${THRIFT_GEN_CPP_DIR}/${THRIFT_NAME}_constants.h
+    ${THRIFT_GEN_CPP_DIR}/${THRIFT_NAME}_types.cpp
+    ${THRIFT_GEN_CPP_DIR}/${THRIFT_NAME}_types.h
+    ${THRIFT_GEN_CPP_DIR}/Standard.cpp
+    ${THRIFT_GEN_CPP_DIR}/Standard.h
```

Comment:
These files will appear multiple times in the ALL_THRIFT_CPP_FILES var (once for each iteration). Should we care? Probably not too dangerous as it is only being used in the add_library call.